warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelper.java

```diff
@@ -0,0 +1,247 @@
+ * </p>
+ */
+public class WhindexFieldIngestHelper implements WhindexIngest {
+
+    // Logger for logging warnings and debug information.
+    private static final Logger log = Logger.getLogger(WhindexFieldIngestHelper.class);
+
+    // Constants defining property names used in the configuration.
+    public static final String WHINDEX_RULES = "whindex.rules";
+    public static final String VALUE_FIELD = "value_field";
+    public static final String SRC_FIELD = "src_field";
+    public static final String DELETE_SRC_FIELD = "delete_src_field";
+    public static final String DST_FIELD = "dst_field";
+    public static final String VALUES = "values";
+
+    // The Type object representing the context of ingestion.
+    private final Type type;
+    // Multimap mapping value field names to their corresponding WhindexConfig instances.
+    private final Multimap<String,WhindexConfig> valueFieldsToWhindexConfigs = HashMultimap.create();
+    // Set of source fields that are marked as overloaded (to be deleted after processing).
+    private final Set<String> overloadedFields = new HashSet<>();
+    // Set of destination (whindex) fields generated from configuration rules.
+    private final Set<String> destinationFields = new HashSet<>();
+
+    /**
+     * Constructs a new WhindexFieldIngestHelper for the given ingestion type.
+     *
+     * @param type
+     *            the ingestion type that this helper will operate on.
+     */
+    public WhindexFieldIngestHelper(Type type) {
+        this.type = type;
+    }
+
+    /**
+     * Parses the whindex rules from the provided Hadoop Configuration.
+     * <p>
+     * The configuration is expected to have properties in the following form: <code>typeName.whindex.rules.[groupID].[property]=value</code>, where the
+     * property is one of VALUE_FIELD, SRC_FIELD, DELETE_SRC_FIELD, DST_FIELD, or VALUES. Each groupID represents a separate whindex rule.
+     * </p>
+     *
+     * @param config
+     *            the Hadoop Configuration containing whindex rules.
+     * @throws IllegalArgumentException
+     *             if there are configuration issues.
+     */
+    @Override
+    public void setup(Configuration config) throws IllegalArgumentException {
+        // Construct a common prefix based on the type name and the whindex rules constant.
+        String commonPrefix = type.typeName() + "." + WHINDEX_RULES + ".";
+
+        // Get all properties from the configuration that start with the common prefix.
+        Map<String,String> properties = config.getPropsWithPrefix(commonPrefix);
+        // Map to temporarily group properties by their group ID.
+        Map<String,WhindexConfig> groupingsToConfigs = new HashMap<>();
+
+        // Process each configuration property.
+        for (Map.Entry<String,String> entry : properties.entrySet()) {
+            // Split the key by '.' to extract the group ID and property name.
+            String[] parts = entry.getKey().split("\\.");
```

Comment:
A little validation here.... only 2 parts and neither part is empty/0 length.  Also ensure value is non-empty.